packages/TorneloScoresheet/src/pages/GraphicalRecording/GraphicalRecording.tsx

```diff
@@ -171,18 +173,32 @@ const GraphicalRecording: React.FC = () => {
     : [];
 
   // Button Functions
-  const handleConfirmWinner = (signature: string) => {
+
+  const handleConfirmWinner = (signatureInput: string) => {
+    const signatureArray = [...playerSignatures, signatureInput];
     if (!graphicalRecordingMode || !goToEndGame) {
       return;
     } else {
-      goToEndGame({
-        winner: selectedWinner?.color ?? null,
-        signature: signature,
-        gamePgn: pgn,
-      });
+      if (signatureArray.length == 2) {
+        setPlayerSignatures(() => []);
+        goToEndGame({
+          winner: selectedWinner?.color ?? null,
+          signature: signatureArray,
+          gamePgn: pgn,
+        });
+      }
     }
   };
```

Comment:
And then in here we would be checking which of the signature Record still has an undefined for the signature, set the Signature Component's player to the one who needs to sign, then show the signature popup, the signature component's onConfirm function will be this function again, where the signature is saved and the same check will be performed again for the other player, and once both signature records are not undefined, we can proceed to call the hooks function